src/components/Invoices/InvoiceDetail/InvoiceDetail.tsx

```diff
@@ -51,20 +67,68 @@ export const InvoiceDetail = () => {
         isReadOnly={isReadOnly}
         formState={formState}
         setIsReadOnly={setIsReadOnly}
+        openInvoicePaymentDrawer={() => setIsPaymentDrawerOpen(true)}
       />
     )
   }, [onSubmit, isReadOnly, formState])
 
+  const hasChanges = !isReadOnly && formState.isDirty
+  const onGoBack = useCallback(() => {
+    if (hasChanges) {
+      setIsDiscardChangesModalOpen(true)
+    }
+    else {
+      toInvoiceTable()
+    }
+  }, [hasChanges, toInvoiceTable])
+
   return (
-    <BaseDetailView slots={{ Header }} name='Invoice Detail View' onGoBack={toInvoiceTable}>
-      {viewState.mode === UpsertInvoiceMode.Update && <InvoiceDetailSubHeader invoice={viewState.invoice} />}
-      <InvoiceForm
-        isReadOnly={isReadOnly}
-        onSuccess={onSuccess}
-        onChangeFormState={onChangeFormState}
-        ref={formRef}
+    <>
+      <BaseDetailView slots={{ Header, BackIcon: hasChanges ? X : BackArrow }} name='Invoice Detail View' onGoBack={onGoBack}>
+        {viewState.mode === UpsertInvoiceMode.Update && <InvoiceDetailSubHeader invoice={viewState.invoice} />}
+        <InvoiceForm
+          isReadOnly={isReadOnly}
+          onSuccess={onUpsertInvoiceSuccess}
+          onChangeFormState={onChangeFormState}
+          ref={formRef}
+        />
+      </BaseDetailView>
+      <BaseConfirmationModal
+        isOpen={isDiscardChangesModalOpen}
+        onOpenChange={setIsDiscardChangesModalOpen}
+        title='Discard changes to this invoice?'
+        description='Any unsaved changes will be lost.'
+        onConfirm={toInvoiceTable}
+        confirmLabel='Discard changes'
+        cancelLabel='Keep editing'
       />
-    </BaseDetailView>
+      {viewState.mode === UpsertInvoiceMode.Update && (
+        <Drawer isOpen={isPaymentDrawerOpen} onOpenChange={setIsPaymentDrawerOpen}>
+          {({ close }) => (
+            <VStack pb='lg' gap='lg'>
+              <VStack pi='md'>
+                <ModalTitleWithClose
+                  heading={(
+                    <ModalHeading size='lg'>
+                      Record invoice payment
+                    </ModalHeading>
+                  )}
+                  onClose={close}
+                />
+              </VStack>
```

Comment:
This looks to me like we could stick this into the `Drawer` as a `header` slotProp.